llvm/lib/Target/AMDGPU/SIMemoryLegalizer.cpp

```diff
@@ -10,6 +10,20 @@
 /// Memory legalizer - implements memory model. More information can be
 /// found here:
 ///   http://llvm.org/docs/AMDGPUUsage.html#memory-model
+///
+/// Notes on the Vulkan Memory Model, which is controlled by the
+/// `amdgpu.vulkan.memory-model` module flag. When that flag
+/// is provided:
+/// - vulkan:private operations cannot be atomic, and as such
```

Comment:
"cannot be atomic" is way too imprecise. The exact spec is that "private operations are not synchronized in a running shader and hence they can always be cached".